### ISSUE TITLE
Documentation Reorg / Cleanup - Meta Issue

### ISSUE THREAD

**Issue by jwrober** [longturn/freeciv21#1367]
There are a few things that I would like to see us do to improve the layout of the documentation (`docs`) as well as how we interface/integrate with the in-game help. I reviewed the documentation for "The Battle of Wesnoth" and "Warzone 2100" for a view in what other popular OSS games are doing.

A view of current status:
* `General/*`: This directory has a miss-mash of content. FAQ, Install & Compile, Windows Install, Modpack Installer usage, server introduction. Sections should be on topic.
* `General/Manuals/client-manual`: Client specific manual. Other games that are technically client/server obfuscate the difference between the two and keep the conversation higher level for base content. Wesnoth differentiates between regular (e.g. casual) players vs advanced players.
* `Playing/*`: We have good docs here that are topic specific. The How To Play page has content that is very close, but also different in many ways as the "Strategy and Tactics" page in the in-game help. The topic of the Citizen Governor is on topic and easily linked to from other pages.
* `Modding/*`: On topic, well laid out.
* `Contributing/*`: Also on topic with some very recent improvements/changes.
* `Coding/*`: On topic as well. Similar to `Contributing`, has some very recent changes with some pending issues.

A potential future state:
* If a reader would read from "top to bottom", we should start with an "about fc21" page, then lead into "getting fc21" and eventually "compiling" it as a last resort. See how Wesnoth does it at https://wiki.wesnoth.org/Play. We sort of have a beginning of this on the main repo `README.md` file, but even that is incomplete.
* We need to break out and in some cases consolidate duplicate content. The FAQ, How to Play, Client Manual and "Strategy and Tactics" in-game help all have duplicate text. This should be consolidated as much as possible. The FAQ has a lot of super advanced stuff in it that should be broken out.
* We need to have "casual user" and "advanced user" documents. The casual user section should obfuscate the client/server nature of the game. You just "play". The client manual should be broken apart into casual and advanced concepts with cross-links where necessary. The advanced user section would get into the more technical aspects of the game, especially the server and hosting multiplayer games. The `on-the-server.rst` file is better in advanced user for example. 
* The current Client Manual should be broken apart and turned into a Game Manual. When talking about the two "options" dialog boxes: the current "local options" dialog should become the "interface options" and the "server options" should become the "game options".

So how do we break this up into small-ish PRs to aid reviews?

- [x] I think we need to change `General` to `Getting` and move things around. 
    - [x] Keep `install.rst`, and `windows-install.rst`. 
    - [x] Add an `about.rst` to better introduce Freeciv21 and add some screenshots of some of our more epic LT games. Maybe include some of the animated GIF's as well. For `about.rst` use content from the "Introduction" in-game help topic as they should be very similar. 
    - [x] Potentially update the `README.md` to keep introduction text consistent.
    - [x] Move `faq.rst` to `Playing`
    - [x] Move `modpack-installer.rst` to `Getting/Manuals`
    - [x] Move `on-the-server.rst` to `Getting/Manuals`
    - [x] Break out `install.rst` to an Installation and Compiling files. Update list of Linux OS's we support for Qt 5.15+

- [x] Move `Getting/Manuals` to `Manuals`

- [x] De-Duplicate content in `how-to-play.rst` and "Strategy and Tactics" in-game help topic.

- [x] In multiple PRs, break out `client-manual.rst` and `on-the-server.rst` into "Game Manual" and "Advanced Game Topics". Take advantage of links to the new `how-to-play.rst`.

- [x] Create `Manuals/Game` and `Manuals/Advanced` directories and then move `on-the-server.rst` to `General/Advanced`
    - [x] Break out `client-manual.rst` into multiple pages/files into both `Game` and `Advanced` directories. For example, right now the super detailed description of the Start New Game dialog is better in `Advanced`. The `Game` version would be a much simpler description. (this one might end up being a large PR)
    - [x] Greatly reduce the size of the FAQ by moving questions to pages associated with either the Game Manual or Advanced Game Topics (maybe something like advanced recipes). This could be done across many PRs.
    - [x] Create a Glossary from the FAQ and other pages

Once this is done, we could tackle pretty much all of the issues below, especially the `helpdata.txt` static content merge into a more extensive "how to play" manual. Hopefully we will have some of the rulesets fully documented by then to aid cross links.

Related Issues:
* #481
* #506
* #1232
* #1240
* #1264
* #1318
* #1321
* #1336
